### PR TITLE
get system information 

### PR DESCRIPTION
**Pull request by antl31**
Succesfull build: https://travis-ci.org/github/reportportal/client-Python/jobs/667156824

### CODE REVIEW

File: reportportal_client/service.py [hunk 1]
```diff
@@ -426,3 +427,27 @@ def log_batch(self, log_data, item_id=None):
         logger.debug("log_batch response: %s", r.text)
 
         return _get_data(r)
+
+    @staticmethod
+    def get_system_infromation():
```
Comment (iivanou) [inline reportportal_client/service.py]:
```    
    @staticmethod
    def get_system_infromation(agent_name='agent_name'):
        """Get system information about agent, os, cpu, system, etc.

        :param agent_name: Name of the agent: pytest-reportportal, ..., ...
        :return: dict      HERE IS EXAMPLE OF THE DICT
        """
        try:
            agent_version = pkg_resources.get_distribution(agent_name)
            agent = '{0}-{1}'.format(agent_name, agent_version)
        except pkg_resources.DistributionNotFound:
            agent = 'not found'

         return = {'agent': agent,
                   'os': platform.system(),
                   'cpu': platform.processor(),
                   'machine': platform.machine()}
```

File: tests/test_service.py [hunk 2]
```diff
@@ -82,3 +83,49 @@ def test_finish_launch(self, mock_get, rp_service):
...
+
+    @mock.patch('platform.system')
+    @mock.patch('platform.machine')
+    @mock.patch('platform.processor')
+    @mock.patch('pkg_resources.get_distribution')
+    @mock.patch('pkg_resources.Distribution')
+    def test_get_system_information(self, distribution_mock,
+                                    get_distribution_mock, processor_mock,
+                                    machine_mock, system_mock):
+        """
```
Comment (iivanou) [inline tests/test_service.py]:
```suggestion
        """Test for validate get_system_information.
```

File: tests/test_service.py [hunk 3]
```diff
@@ -82,3 +83,49 @@ def test_finish_launch(self, mock_get, rp_service):
         mock_get.return_value = {"id": 111}
         _get_msg = rp_service.finish_launch('name', datetime.now().isoformat())
         assert _get_msg == {"id": 111}
+
+    @mock.patch('platform.system')
+    @mock.patch('platform.machine')
+    @mock.patch('platform.processor')
+    @mock.patch('pkg_resources.get_distribution')
+    @mock.patch('pkg_resources.Distribution')
```
Comment (iivanou) [inline tests/test_service.py]: Is not required to be mocked.

File: tests/test_service.py [hunk 4]
```diff
@@ -82,3 +83,49 @@ def test_finish_launch(self, mock_get, rp_service):
...
+    def test_get_system_information(self, distribution_mock,
+                                    get_distribution_mock, processor_mock,
+                                    machine_mock, system_mock):
+        """
+        Test for validate get_system_information.
+
+        :param distribution_mock: Mock object of Distribution class
```
Comment (iivanou) [inline tests/test_service.py]: Please, align parameter description by the longest one.

File: tests/test_service.py [hunk 5]
```diff
@@ -82,3 +83,49 @@ def test_finish_launch(self, mock_get, rp_service):
         mock_get.return_value = {"id": 111}
         _get_msg = rp_service.finish_launch('name', datetime.now().isoformat())
         assert _get_msg == {"id": 111}
+
+    @mock.patch('platform.system')
+    @mock.patch('platform.machine')
+    @mock.patch('platform.processor')
```
Comment (iivanou) [inline tests/test_service.py]:
```
@patch('platform.processor', Mock(return_value='amd'))
def test_platform():
    assert platform.processor() == 'amd'
```

File: reportportal_client/service.py (hunk 1 above)
Comment (iivanou) [inline reportportal_client/service.py]: Typo: get_system_infORmation

Comment (iivanou) [inline reportportal_client/service.py]: Let's use this method and generate attributes for the launch.

File: reportportal_client/service.py [hunk 6]
```diff
@@ -42,14 +43,15 @@ def _convert_string(value):
...
 
 
-def _list_to_payload(dictionary):
+def _dict_to_payload(dictionary):
     """Convert dict to list of dicts.
 
     :param dictionary: initial dict
     :return list: list of dicts
     """
+    system = dictionary.pop('system', False)
```
Comment (iivanou) [inline reportportal_client/service.py]: Use either single quotes or doubles within the whole module.

File: reportportal_client/service.py [hunk 7]
```diff
@@ -426,3 +428,24 @@ def log_batch(self, log_data, item_id=None):
         logger.debug("log_batch response: %s", r.text)
 
         return _get_data(r)
+
+    @staticmethod
+    def get_system_information(agent_name='agent_name'):
+        """Get system information about agent, os, cpu, system, etc.
+
+        :param agent_name: Name of the agent: pytest-reportportal, ..., ...
```
Comment (iivanou) [inline reportportal_client/service.py]: I thought you will fill up the list of agent names:)

File: tests/test_service.py [hunk 8]
```diff
@@ -1,19 +1,19 @@
 """This modules includes unit tests for the service.py module."""
 
-
 from datetime import datetime
 from six.moves import mock
-
+from pkg_resources import DistributionNotFound
```
Comment (iivanou) [inline tests/test_service.py]: https://www.python.org/dev/peps/pep-0008/#imports

Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
You should put a blank line between each group of imports.


Also, please use alphabetical order inside of the group.

File: tests/test_service.py [hunk 9]
```diff
@@ -26,12 +26,19 @@ def test_check_convert_to_string(self):
         expect(lambda: isinstance(_convert_string("Hello world"), str))
         assert_expectations()
 
-    def test_list_to_payload(self):
-        """Test convert dict to list of dicts."""
-        initial_dict = {'key': "value", 'key1': 'value1'}
-        expected_list = [{'key': 'key', 'value': 'value'},
-                         {'key': 'key1', 'value': 'value1'}]
-        assert _list_to_payload(initial_dict) == expected_list
+    def test_dict_to_payload_without_system_key(self):
```
Comment (iivanou) [inline tests/test_service.py]: These two tests can be combined into the one function wrapped with the parametrize marker.

```
    @pytest.mark.parametrize('system', [True, False])
    def test_dict_to_payload_with_system_key(self, system):
        """Test convert dict to list of dicts with key system."""
        initial_dict = {"aa": 1, "b": 2, "system": system}
        expected_list = [{'key': 'aa', 'value': '1', 'system': system},
                         {'key': 'b', 'value': '2', 'system': system}]
        assert _dict_to_payload(initial_dict) == expected_list
```

File: tests/test_service.py [hunk 10]
```diff
@@ -82,3 +89,35 @@ def test_finish_launch(self, mock_get, rp_service):
...
+                mock.Mock(return_value='pytest 5.0'))
+    def test_get_system_information(self):
+        """Test for validate get_system_information."""
+
+        expected_result = {'agent': 'pytest-pytest 5.0',
+                           'cpu': 'amd',
+                           'machine': 'Windows-PC',
+                           'os': 'linux'}
+
+        assert ReportPortalService.get_system_information('pytest') \
```
Comment (iivanou) [inline tests/test_service.py]: cond = (ReportPortalService.get_system_information('pytest') == expected_result)
assert cond

File: reportportal_client/service.py (hunk 6 above)
Comment (antl31) [reply reportportal_client/service.py]: it doesnt matter, pep8 allow to use single and double quotes

File: reportportal_client/service.py [hunk 11]
```diff
@@ -426,3 +428,26 @@ def log_batch(self, log_data, item_id=None):
         logger.debug("log_batch response: %s", r.text)
 
         return _get_data(r)
+
+    @staticmethod
+    def get_system_information(agent_name='agent_name'):
+        """Get system information about agent, os, cpu, system, etc.
```
Comment (iivanou) [inline reportportal_client/service.py]:
```
        """Get system information about agent, os, cpu, system, etc.

        :param agent_name: Name of the agent: pytest-reportportal,
                           roborframework-reportportal, nosetest-reportportal,
                           behave-reportportal
        :return:           dict {'agent': pytest - 5.0.5,
                                 'os': 'Windows',
                                 'cpu': 'AMD',
                                 'machine': "Windows10_pc"}
        """
```

File: tests/test_service.py [hunk 12]
```diff
@@ -1,19 +1,21 @@
 """This modules includes unit tests for the service.py module."""
 
-
 from datetime import datetime
```
Comment (iivanou) [inline tests/test_service.py]:
```
from datetime import datetime
from pkg_resources import DistributionNotFound

from delayed_assert import expect, assert_expectations
import pytest
from six.moves import mock

from reportportal_client.service import (
    ReportPortalService,
    _convert_string,
    _dict_to_payload,
    _get_data,
    _get_id,
    _get_json,
    _get_messages,
    _get_msg,
    _list_to_payload
)
```

**Pull request merged**